x/concentrated-liquidity/fees.go

```diff
@@ -196,39 +196,22 @@ func (k Keeper) getInitialFeeGrowthOutsideForTick(ctx sdk.Context, poolId uint64
 
 // collectFees collects the fees earned by a position and sends them to the owner's account.
 // Returns error if the position with the given id does not exist or if fails to get the fee accumulator.
-func (k Keeper) collectFees(ctx sdk.Context, owner sdk.AccAddress, positionId uint64) (sdk.Coins, error) {
-	// Get the position with the given ID.
+func (k Keeper) collectFees(ctx sdk.Context, sender sdk.AccAddress, positionId uint64) (sdk.Coins, error) {
 	position, err := k.GetPosition(ctx, positionId)
 	if err != nil {
 		return sdk.Coins{}, err
 	}
 
-	// Get the fee accumulator for the position's pool.
-	feeAccumulator, err := k.getFeeAccumulator(ctx, position.PoolId)
+	isOwner, err := k.isPositionOwner(ctx, sender, position.PoolId, positionId)
 	if err != nil {
 		return sdk.Coins{}, err
 	}
 
-	// Get the key for the position's accumulator in the fee accumulator.
-	positionKey := types.KeyFeePositionAccumulator(positionId)
-
-	// Check if the position exists in the fee accumulator.
-	hasPosition, err := feeAccumulator.HasPosition(positionKey)
-	if err != nil {
-		return sdk.Coins{}, err
-	}
-	if !hasPosition {
-		return sdk.Coins{}, types.PositionIdNotFoundError{PositionId: positionId}
+	if !isOwner {
+		return sdk.Coins{}, types.NotPositionOwnerError{Address: sender.String(), PositionId: positionId}
 	}
 
-	// Compute the fee growth outside of the range between the position's lower and upper ticks.
-	feeGrowthOutside, err := k.getFeeGrowthOutside(ctx, position.PoolId, position.LowerTick, position.UpperTick)
-	if err != nil {
-		return sdk.Coins{}, err
-	}
-
-	// Prepare the position's accumulator for claiming rewards and claim the rewards.
-	feesClaimed, _, err := prepareAccumAndClaimRewards(feeAccumulator, positionKey, feeGrowthOutside)
```

Comment:
I am personally highly against removing comments like these. It allows me to quickly skim the intent of each line, then re-read and check if the function is doing what each line intended.

Will not block on this as its clearly a preference thing, but I will forever write code in this way.

Reply:
Note that I did not remove any comments that still apply. I just did not add any comments for the new logic since it's trivial

I reduced code duplication by introducing an unexported getClaimableFees. So the old comments do not apply here anymore.

Prior to this change, GetClaimableFees and CollectFees were essentially the same with the exception of cache context